File: rabbitmq-sample/src/main/java/yun/blog/rabbitmqsample/member/Member.java
package yun.blog.rabbitmqsample.member;


import java.time.LocalDateTime;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

@Entity
@Table(name = "member")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Getter
public class Member {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private long id;

  @Column(name = "name", nullable = false)
  private String name;

  @Column(name = "email", nullable = false)
  private String email;

  @CreationTimestamp
  @Column(name = "create_at", nullable = false, updatable = false)
  private LocalDateTime createAt;

  @UpdateTimestamp
  @Column(name = "update_at", nullable = false)
  private LocalDateTime updateAt;

  @Builder
  public Member(String name, String email) {
    this.name = name;
    this.email = email;
  }
}

File: spring-kotlin/src/main/kotlin/com/example/springkotlin/member/Member.kt
package com.example.springkotlin.member

import javax.persistence.*


@Entity
@Table(name = "member")
@Access(AccessType.FIELD) // 용도는 ?
class Member protected constructor() {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0

    @Column(name = "email", nullable = false, unique = true)
    lateinit var email: String
        protected set

    @Column(name = "name", nullable = false, unique = true)
    lateinit var name: String
        protected set

    constructor(email: String, name: String) : this() {
        this.email = email
        this.name = name
    }


}